MobSF/utils.py

```diff
@@ -59,8 +60,17 @@ def check_update():
     try:
         print("\n[INFO] Checking for Update.")
         github_url = "https://raw.githubusercontent.com/MobSF/Mobile-Security-Framework-MobSF/master/MobSF/settings.py"
-        response = urllib.request.urlopen(github_url)
-        html = str(response.read()).split("\n")
+        if settings.UPSTREAM_PROXY_ENABLED:
+           if not settings.UPSTREAM_PROXY_USERNAME :
+               proxy_port = str(settings.UPSTREAM_PROXY_PORT)
+               proxy_host = settings.UPSTREAM_PROXY_TYPE + '://'  + settings.UPSTREAM_PROXY_IP + ':' + proxy_port
+               proxies = {"https": proxy_host}
+           else:
+               proxy_port = str(settings.UPSTREAM_PROXY_PORT)
+               proxy_host = settings.UPSTREAM_PROXY_TYPE + '://' + settings.UPSTREAM_PROXY_USERNAME + ':' + settings.UPSTREAM_PROXY_PASSWORD + "@" + settings.UPSTREAM_PROXY_IP + ':' + proxy_port
+               proxies = {"https": proxy_host}
+        response = requests.get(github_url, proxies=proxies)

```

Comment:
This will surely fail if `UPSTREAM_PROXY_ENABLED = False` as `proxies` will be not in scope. Unit tests @superpoussin22 with proxy enabled and disabled :)

Reply:
This is why I suggested move this as a single function and use it from other places.

Reply:
th'at's in my to-do after strings problem